doc/SAI-Proposal-versioning.md

```diff
@@ -0,0 +1,74 @@
+SAI Versioning 
+==============
+
+This document describes various proposals to include versioning in the SAI library. 
+
+Requirements
+------------
+Provide a mechanism for vendors to find the current version of SAI headers at compile time and at runtime if it is warranted. 
+
+Currently, if a vendor wants to support multiple versions of SAI, 
+
+### Binaries 
+A vendor publishes different libSAI implementations each compatible with different SAI versions. The users of the binaries can choose a version of the binary depending on which version of SAI they use for Sonic. The vendor usually publishes these binaries from different development branches dedicated to each SAI version. 
+
+### Source Code 
+A vendor publishes their SDK in source form and advice their users of which version of their SDK can be used for the requested SAI version. It is assumed the most common form this takes is each vendor maintains a trunk branch with the latest code and patches are backported to specific branches created for each SAI version. 
+
+> ### Example 
+> Customer currently using Sonic version 201911 which is compatible with SAI 1.5.4. The vendor SDK version is 1.1. They customer wants to upgrade to a newer version of the SDK which supports SAI objects/attributes that are only supported in 1.2. The common method to support this is to backport the supported attributes to 1.1.  
+
+This proposal simply wishes to solve the following question. 
+**How does a vendor support multiple SAI versions using only the trunk branch?**
+
+Proposal 1
+----------
+This proposal provides a static configuration file with version numbers. 
+ 
+### version.h 
```

Comment:
is this version.h in SAI repo ? or vendor repo ?

Reply:
version.h and version.mk will go into SAI. A vendor/user can use data from these files when building code.
This is just sample code I copied from somewhere. static will go away.

Reply:
changed to #define.